CppCoreGuidelines.md

```diff
@@ -6903,43 +6903,57 @@ not using this (over)general interface in favor of a particular interface found
 ##### Reason
 
 A class is more stable (less brittle) if it does not contain data.
-Interfaces should normally be composed entirely of public pure virtual functions and a default/empty virtual destructor.
+An interface should be composed entirely of pure virtual functions and a virtual destructor; see [C.35](#Rc-dtor-virtual).
+An interface should have a defaulted default constructor.
 
 ##### Example
 
-    class My_interface {
+    class ClassicalConnection {  // good
     public:
-        // ...only pure virtual functions here ...
-        virtual ~My_interface() {}   // or =default
+        virtual std::string read() = 0;
+        virtual void write(std::string_view) = 0;
+        virtual ~ClassicalConnection() = default;
     };
 
-##### Example, bad
-
-    class Goof {
+    class NVIConnection {  // also good
```

Comment:
maybe keep just one? The protected: section in this one may be distracting.

Reply:
Hmm. We could just remove the line `protected:`, making those members public; that's what I'd do in production code anyway.

I'm loath to show a lot of examples of polymorphic hierarchies in the pre-NVI, Java-flavored style; people might get the idea that the Guidelines are in favor of that style. For this rule in particular I want to clarify that NVI does not conflict with the Guidelines.

Reply:
The guidelines are actually not in favor of either. I tried introducing NVI  as a rule in #768

Reply:
Editors call: Could you please remove the NVI example?